packages/headless/src/api/service/insight/user-actions/user-actions-request.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  baseInsightRequest,
+  InsightParam,
+  pickNonInsightParams,
+} from '../insight-params';
+
+export type InsightUserActionsRequest = InsightParam &
+  TicketCreationDateParam &
+  NumberSessionsBeforeParam &
+  NumberSessionsAfterParam &
+  MaximumSessionInactivityMinutesParam &
+  ExcludedCustomActionsParam;
+
+interface TicketCreationDateParam {
```

Comment:
Feel free to do what you think is best for your use case, but you don't necessarily need to"explode" all parameters like that in various interface.

This is a pattern we use for the `search` params because they are re-used for various use cases (eg: recommendation, product listing). So in that case it's nice to be able to pick and compose different parameters together to build a new type of Request without duplication.

If those parameters are only used for a single route in the Insight Panel API, and not re-used for other services/use cases, then that "explosion" of interface might be overkill.